File: Samples/get_set_formulas_results_csharp_vs2008/get_set_formulas_results_csharp_vs2008/shape_getformulas.cs
using IVisio = Microsoft.Office.Interop.Visio;

public static partial class CSharpSamples
{
    public static void Shape_GetFormulas(IVisio.Document doc)
    {
        var pages = doc.Pages;
        var page = pages.Add();
        page.NameU = "SGF";

        var shape = page.DrawRectangle(1, 1, 4, 3);
        shape.get_CellsU("Width").Formula = "=(1.0+2.5)";
        shape.get_CellsU("Height").Formula = "=(0.0+1.5)";

        // BUILD UP THE REQUEST
        var items = new[]
        {
            new {   section = (short) IVisio.VisSectionIndices.visSectionObject, 
                    row     = (short) IVisio.VisRowIndices.visRowXFormOut, 
                    cell    = (short) IVisio.VisCellIndices.visXFormWidth },
            
            new {   section = (short) IVisio.VisSectionIndices.visSectionObject, 
                    row     = (short) IVisio.VisRowIndices.visRowXFormOut, 
                    cell    = (short) IVisio.VisCellIndices.visXFormHeight }
        };

        // MAP THE REQUEST TO THE STRUCTURES VISIO EXPECTS
        short[] SRCStream = new short[items.Length * 3];
        for (int i = 0; i < items.Length; i++)
        {
            SRCStream[i * 3 + 0] = items[i].section;
            SRCStream[i * 3 + 1] = items[i].row;
            SRCStream[i * 3 + 2] = items[i].cell;
        }

        // EXECUTE THE REQUEST
        System.Array formulas_sa;
        System.Array SRCStream_sa = (System.Array) SRCStream;
        shape.GetFormulasU(ref SRCStream_sa, out formulas_sa);

        // MAP OUTPUT BACK TO SOMETHING USEFUL 
        object [] formulas_objects = (object[])formulas_sa;
        string [] formulas = new string[formulas_objects.Length];
        formulas_objects.CopyTo(formulas, 0);

        // DISPLAY THE INFORMATION
        shape.Text = string.Format("Formulas={0},{1}", formulas[0], formulas[1]);
    }
}

File: Samples/visio_managed_code_interop_vs2010/VS2010_IronPython_Samples/Shape_GetFormulas.py
import clr 
import System 

clr.AddReference("Microsoft.Office.Interop.Visio") 
import Microsoft.Office.Interop.Visio 
IVisio = Microsoft.Office.Interop.Visio 

from Records import *
import Util

def Shape_GetFormulas( doc ):

        pages = doc.Pages
        page = pages.Add()
        page.NameU = "SGF"
        
        shape = page.DrawRectangle(1, 1, 4, 3)
        shape.CellsU["Width"].Formula = "=(1.0+2.5)"
        shape.CellsU["Height"].Formula = "=(0.0+1.5)"

        # BUILD UP THE REQUEST
        items = [
                Shape_GetFormulas_Record(IVisio.VisSectionIndices.visSectionObject, IVisio.VisRowIndices.visRowXFormOut, IVisio.VisCellIndices.visXFormWidth),
                Shape_GetFormulas_Record(IVisio.VisSectionIndices.visSectionObject, IVisio.VisRowIndices.visRowXFormOut, IVisio.VisCellIndices.visXFormHeight)
        ]

        # MAP THE REQUEST TO THE STRUCTURES VISIO EXPECTS
        SRCStream = Util.get_new_system_array(System.Int16, len(items)*3)
        for i in xrange(len(items)) :
                SRCStream[i * 3 + 0] = items[i].SectionIndex
                SRCStream[i * 3 + 1] = items[i].RowIndex
                SRCStream[i * 3 + 2] = items[i].CellIndex

        # EXECUTE THE REQUEST
        formulas_sa = Util.get_outref_to_system_array(System.Object) 
        SRCStream_sa = Util.get_ref_to_system_array(System.Int16,SRCStream)  
        shape.GetFormulasU(SRCStream_sa, formulas_sa)

        # OUTPUT BACK TO SOMETHING USEFUL 
        formulas = Util.get_new_system_array(System.String,formulas_sa.Length)
        formulas_sa.CopyTo(formulas, 0);

        shape.Text = System.String.Format("Formulas={0},{1}", formulas[0], formulas[1])

